### PR TITLE
feat(transactions): Start processing `transaction_source` in transactions_processor.py

### PR DESCRIPTION
**Pull request by andriisoldatenko**
Start processing new transaction annotation:

```
{ "transaction_info": {"source": "url"} }
```

https://develop.sentry.dev/sdk/event-payloads/transaction/#transaction-annotations

and storing  it using new column `transaction_source` from https://github.com/getsentry/snuba/pull/2982

**PLEASE DON'T MERGE** until https://github.com/getsentry/snuba/pull/2982

resolves [TET-270]

[TET-270]: https://getsentry.atlassian.net/browse/TET-270?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: snuba/datasets/transactions_processor.py [hunk 1]
```diff
@@ -110,6 +110,9 @@ def _process_base_event_values(
         processed["transaction_name"] = _unicodify(
             event_dict["data"].get("transaction") or ""
         )
+        processed["transaction_source"] = _unicodify(
+            event_dict["data"].get("transaction_info", {}).get("source") or ""
```
Comment (jan-auer) [inline snuba/datasets/transactions_processor.py:114]: Just to be extra careful, could we write it like this?

```suggestion
            (event_dict["data"].get("transaction_info") or {}).get("source") or ""
```

That would also cover the case where the dict contains an explicit `None`.

**Comment by andriisoldatenko** [comment]
> Looks good. I may have forgotten to mention, in order to expose this column in the data model, you will need to add a column here: https://github.com/getsentry/snuba/blob/master/snuba/datasets/storages/transactions_common.py#L40

Done.

**Review by evanh — APPROVED** [review]

**Review by nikhars — APPROVED** [review]
The migration has been applied to production. Please merge and deploy during SnS working hours (PST).

**Comment by andriisoldatenko** [comment]
@nikhars  are we good to merge it today?

**Comment by nikhars** [comment]
> @nikhars are we good to merge it today?

Let me do a sanity check of the clusters to ensure the migration has happened on both the clusters. Want to be cautious since there are 2 clusters right now.

**Comment by nikhars** [comment]
> > @nikhars are we good to merge it today?
> 
> Let me do a sanity check of the clusters to ensure the migration has happened on both the clusters. Want to be cautious since there are 2 clusters right now.

Verified. Feel free to merge now.

**Comment by andriisoldatenko** [comment]
@nikhars  done!

**Pull request merged**